docs/developer/tutorials/snmp/profile-format.md

```diff
@@ -0,0 +1,204 @@
+### `sysobjectid`
+
+_(Required)_
+
+The `sysobjectid` field is used to match profiles against devices during device autodiscovery.
+
+It can refer to a fully-defined OID for a specific device make and model:
+
+```yaml
+sysobjectid: 1.3.6.1.4.1.232.9.4.10
+```
+
+or a wildcard pattern to address multiple device models:
+
+```yaml
+sysobjectid: 1.3.6.1.131.12.4.*
+```
+
+### `extends`
+
+_(Optional)_
+
+This field can be used to include metrics and metric tags from other so-called _base profiles_. Base profiles can derive from other base profiles to build a hierarchy of reusable profile mixins.
+
+!!! important
+    All device profiles should extend from the `_base.yaml` profile, which defines items that should be collected for all devices.
+
+Example:
+
+```yaml
+extends:
+  - _base.yaml
+  - _generic-router-if.yaml  # Include basic metrics from IF-MIB.
+```
+
+### `metrics`
+
+_(Required)_
+
+Entries in the `metrics` field define which metrics will be collected by the profile. They can reference either a single OID (a.k.a _symbol_), or an SNMP table.
+
+#### Symbol metrics
+
+In profiles, OIDs can be specified as entries containing the `symbol` field:
+
+```yaml
+metrics:
+  - MIB: TCP-MIB
+    symbol:
+      OID: 1.3.6.1.2.1.6.5
+      name: tcpActiveOpens
+```
+
+#### Table metrics
+
+An SNMP table is an object that is composed of multiple entries.
+
+In a MIB file, tables be recognized by the following syntax:
+
+```asn
+exampleTable OBJECT-TYPE
+    SYNTAX   SEQUENCE OF exampleEntry
+    -- ...
+
+exampleEntry OBJECT-TYPE
+   -- ...
+   ::= { exampleTable 1 }
+```
+
+In profiles, tables can be specified as entries containing the `table` and `symbols` fields:
+
+```yaml
+metrics:
+  - MIB: CISCO-PROCESS-MIB
+    table:
+      # Identification of the table which metrics come from.
+      # For example, this device has multiple CPU units; each row
+      # in this table contains information about a single CPU unit.
+      OID: 1.3.6.1.4.1.9.9.109.1.1.1
+      name: cpmCPUTotalTable
+    symbols:
+      # List of symbols ('columns') to retrieve.
+      # Same format as for a single OID.
+      # Each row in the table will emit these metrics.
+      # For example, if we have N CPU units, then the
+      # integration will emit N `snmp.cpmCPUMemoryUsed` metrics.
+      - OID: 1.3.6.1.4.1.9.9.109.1.1.1.1.12
+        name: cpmCPUMemoryUsed
+      # ...
+```
+
+#### Table metrics tagging
+
+It is possible to add tags to metrics retrieved from a table in three ways:
+
+- Using an index:
+
+```yaml
+metrics:
+  - MIB: CISCO-PROCESS-MIB
+    table:
+      OID: 1.3.6.1.4.1.9.9.109.1.1.1
+      name: cpmCPUTotalTable
+    symbols:
+      - OID: 1.3.6.1.4.1.9.9.109.1.1.1.1.12
+        name: cpmCPUMemoryUsed
+    metric_tags:
+      # Add a 'cpu:<row_index>' tag to each metric of each row,
+      # whose value is the index of the row in the table.
+      # This allows querying metrics for a given CPU unit, e.g. 'cpu:1'.
+      - tag: cpu
+        index: 1
```

Comment:
I went through the rabbit hole after that doc, which made me understand something I completely missed. If you look at a table entry definition in a mib you can for example see something like this:

```
ipIfStatsEntry OBJECT-TYPE
    SYNTAX     IpIfStatsEntry
    MAX-ACCESS not-accessible
    STATUS     current
    DESCRIPTION
           "An interface statistics entry containing objects for a
            particular interface and version of IP."
    INDEX { ipIfStatsIPVersion, ipIfStatsIfIndex }
    ::= { ipIfStatsTable 1 }
```

The import part is the INDEX definition. This will mean that each row with then be suffixed by ipIfStatsIPVersion and then ipIfStatsIfIndex. That allows specific cells to be queries by a GET request when you don't want the whole table.

In our case though, it doesn't matter. All of our indexes usage could be replaced by column tags.

There is one import caveat though: index tagging supports mapping (for ipversion mostlly). If we replace it by the corresponding column, we lose the nice string and get the corresponding value from the enum instead.